### ISSUE TITLE
[Model and README Enhancement] Mushroom Classification using Deep Learning

### ISSUE THREAD

**Issue by Arihant-Bhandari** [abhisheks008/DL-Simplified#651]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Mushroom Classification using Deep Learning</br>
:red_circle: **Aim** : Bettering models by adding keras implemented CNN and CNN with attention models</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/lizhecheng/mushroom-classification</br>
:red_circle: **Approach** : The original author used MAE as his loss value and implemented models, i will be trying out keras based CNN models with categorical cross entropy loss turning the problem into multiclass classification.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Arihant Bhandari
- GitHub Profile Link : https://github.com/Arihant-Bhandari
- Email ID : arihant.bhandari.2004@gmail.com
- Participant ID (if applicable): Arihant_Bhandari (eternal_insight)
- Approach for this Project : adding 2 model implementations under multiclass classification
- What is your participant role? (Mention the Open Source program)
GSSoC Contributor 2024
*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by Arihant-Bhandari** [comment]
hi @abhisheks008 please assign me this, i will also be working on the project idea i asked you about, as soon as i finish with some basic results i will create issue for it as well.

**Comment by abhisheks008** [comment]
The existing project is consisting CNN and Inception for this project, what are models you are planning for this to enhance the accuracy? Please be specific with the architecture names/models.

**Comment by Arihant-Bhandari** [comment]
hi @abhisheks008 i wanted to know how i could turn this issue into level3, from what i gather based on the previous work, the dataset yields very low results , and i myself face similar issues, my best went to about 0.3 % accuracy on baseline CNN, so what i wanted to ask is, if i were to colour train the model, say 3 models whose outputs are voted on for each colour channel as part of my submission alongside 2-4 pretrained models, would this qualify for a level3 contribution ?

**Comment by abhisheks008** [comment]
> hi @abhisheks008 i wanted to know how i could turn this issue into level3, from what i gather based on the previous work, the dataset yields very low results , and i myself face similar issues, my best went to about 0.3 % accuracy on baseline CNN, so what i wanted to ask is, if i were to colour train the model, say 3 models whose outputs are voted on for each colour channel as part of my submission alongside 2-4 pretrained models, would this qualify for a level3 contribution ?

Push your code and let me review it. Will definitely let you know if that qualifies for the level 3.

**Comment by Arihant-Bhandari** [comment]
hi @abhisheks008 i can send in preliminary work, i devised a custom data collection mechanism since the original data owner noted that there are issues with the daatset and people working on EDA concurred with following issues: the images had some duplicates some dark and some grayscaled images when the dataset was supposed to be purely RBG.

alongside this i am also sending in 5 model's work based on how the original author of the repo did his work: i have implemented RESNET50, VGG16, Xception, DenseNet and Inception-ResNet-v2.

in addtion i did a baseline CNN model in keras , which had highest accuracy in all cases as part of followup towards attention based model.

**Comment by abhisheks008** [comment]
> hi @abhisheks008 i can send in preliminary work, i devised a custom data collection mechanism since the original data owner noted that there are issues with the daatset and people working on EDA concurred with following issues: the images had some duplicates, some dark and some grayscaled images when the dataset was supposed to be purely RBG.
> 
> alongside this i am also sending in 5 model's work based on how the original author of the repo did his work: i have implemented RESNET50, VGG16, Xception, DenseNet and Inception-ResNet-v2.
> 
> in addtion i did a baseline CNN model in keras , which had highest accuracy in all cases as part of followup towards attention based model.

Push all your codes together.
